src/super_gradients/common/sg_loggers/dagshub_sg_logger.py

```diff
@@ -0,0 +1,222 @@
+
+        self._init_env_dependency()
+
+        self.log_mlflow_only = log_mlflow_only
+        self.save_checkpoints_dagshub = save_checkpoints_remote
+        self.save_logs_dagshub = save_logs_remote
+
+    @staticmethod
+    def splitter(repo):
+        splitted = repo.split("/")
+        if len(splitted) != 2:
+            raise ValueError(f"Invalid input, should be owner_name/repo_name, but got {repo} instead")
+        return splitted[1], splitted[0]
+
+    def _init_env_dependency(self):
+        """
+        The function creates paths for the DVC directory, models, and artifacts, obtains an authentication token from Dagshub, and sets MLflow tracking credentials. 
+        It also checks whether the repository name and owner have been set and prompts the user to enter them if they haven't. 
+        If the remote URI is not set or does not include "dagshub", Dagshub is initialized with the repository name and owner, and the remote URI is obtained. 
+        The method then creates a Repo object with the repository information and sets the DVC folder to the DVC directory path. 
+        """
+
+        self.paths = {
+            "dvc_directory": Path("artifacts"),
+            "models": Path("models"),
+            "artifacts": Path("artifacts"),
+        }
+
+        token = dagshub.auth.get_token()
+        os.environ["MLFLOW_TRACKING_USERNAME"] = token
+        os.environ["MLFLOW_TRACKING_PASSWORD"] = token
+
+        # Check mlflow environment variable is set:
+        if not self.repo_name or not self.repo_owner:
+            self.repo_name, self.repo_owner = self.splitter(input("Please insert your repository owner_name/repo_name:"))
+
+        if not self.remote or "dagshub" not in os.getenv("MLFLOW_TRACKING_URI"):
+            dagshub.init(repo_name=self.repo_name, repo_owner=self.repo_owner)
+            self.remote = os.getenv("MLFLOW_TRACKING_URI")
+
+        self.repo = Repo(
+            owner=self.remote.split(os.sep)[-2],
+            name=self.remote.split(os.sep)[-1].replace(".mlflow", ""),
+            branch=os.getenv("BRANCH", "main"),
+        )
+        self.dvc_folder = self.repo.directory(str(self.paths["dvc_directory"]))
+
+        mlflow.set_tracking_uri(self.remote)
+        mlflow.set_experiment(self.experiment_name)
+        self.run = mlflow.start_run(nested=True)
+        return self.run
+
+    @multi_process_safe
+    def _dvc_add(self, local_path="", remote_path=""):
+        if not os.path.isfile(local_path):
+            FileExistsError(f"Invalid file path: {local_path}")
+        self.dvc_folder.add(file=local_path, path=remote_path)
+
+    @multi_process_safe
+    def _dvc_commit(self, commit=""):
+        self.dvc_folder.commit(commit, versioning="dvc", force=True)
+
+    @multi_process_safe
+    def add_config(self, tag: str, config: dict):
+        super(DagsHubSGLogger, self).add_config(tag=tag, config=config)
+        param_keys = config.keys()
+        for pk in param_keys:
```

Comment:
Why not:  `mlflow.log_params(config)`?
Wouldn't this work?

Reply:
I did a bit of digging and apparently MLflow doesn't support nested dicts as params. I'll write a flatten function and then log that to MLflow.

Reply:
After looking over this, this would either require changing a lot of parameter names in order to log the 2 broken parameters which are the "list of packages" and the "log schema".

If we were to move to mlflow log_params as is it would break on logging, so for now I suggest we leave it as is. We're looking into a way to fix this on our end.

Reply:
if the issue is the space in the key, we can do 
```
config = {k.replace(" ", "_"):v for k, v in config.items()}
```
before calling `mlflow.log_params(config)`

Reply:
Actually, the problem is that both parameters are too long (since they are big nested params) and MLflow has a character limit for parameter values. So we can either remove them manually, or write code to flatten the entire dict, which makes it less readable.

Happy to do either, though especially since the list of packages will cause the number of parameters to explode.

Reply:
ok. so let's keep it like this for now.